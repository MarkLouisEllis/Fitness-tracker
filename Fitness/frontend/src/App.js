import './App.css';
import {BrowserRouter as Router,Route,Switch} from "react-router-dom";
import React, { useState } from 'react';
import Signup from './components/Signup';
import Welcome from './components/Welcome';
import Login from './components/Login';
import About from './components/About';
import './mycss.css';

//used to check if user has logged in with the createcontext hook. The email,password and admin state is stored in Credential Context
//The provider wrapped around all components at the top level to allows us to use credential state in multiple components. 

export const CredentialsContext = React.createContext();

function  App() {
  
  //state for credentialscontext
  const credentialsState = useState(null);

  return (
    <CredentialsContext.Provider value ={credentialsState}>
      <div className="App">
      <link
           rel="stylesheet"
           href="https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
           integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS"
           crossorigin="anonymous"/>

        <Router>
          <Switch>
            <Route exact path ='/'><Welcome/></Route>
            <Route exact path ='/Signup'><Signup/></Route>
            <Route exact path ='/Login'><Login/></Route>
            <Route exact path ='/About'><About/></Route>
          </Switch>
        </Router>
      </div>
    </CredentialsContext.Provider>  
  );
}

export default App;
