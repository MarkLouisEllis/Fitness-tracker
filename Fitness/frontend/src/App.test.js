import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import React from 'react';



//Snapshot test
//This test is used to establish if the app file with all its components renders correctly and that the UI does not change unexpectedly. 

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//unit test
//I wanted to test the fetch request but I could not get a test to pass

test( 'app fetches' , () => {
expect(App( statusCode )).toBe( 200 );
});
;