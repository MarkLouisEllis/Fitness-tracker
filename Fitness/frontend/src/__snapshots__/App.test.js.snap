// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App"
>
  <link
    crossorigin="anonymous"
    href="https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
    integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS"
    rel="stylesheet"
  />
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        className="container"
      >
        <span
          className="navbar-brand"
        >
          <img
            src="mylogo3.png"
            style={
              Object {
                "width": "3cm",
              }
            }
          />
        </span>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="me-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <a
                href="/About"
                onClick={[Function]}
              >
                About us
              </a>
            </a>
            <a
              className="nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <a
                href="/Signup"
                onClick={[Function]}
              >
                Signup
              </a>
            </a>
            <a
              className="nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <a
                href="/Login"
                onClick={[Function]}
              >
                Login
              </a>
            </a>
          </div>
        </div>
        <div
          aria-expanded={null}
          className="justify-content-end navbar-collapse collapse"
        >
          <span
            className="navbar-text"
          />
        </div>
      </div>
    </nav>
    <h1>
      Welcome 
    </h1>
    <h1>
      To the Fitness Tracker app
    </h1>
    <img
      alt="some picture"
      src="mylogo3.png"
      style={
        Object {
          "width": "8cm",
        }
      }
    />
  </div>
</div>
`;
